src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -690,7 +807,14 @@ ov::npuw::LLMCompiledModel::LLMCompiledModel(const std::shared_ptr<ov::Model>& m
         ov::npuw::ICompiledModel::create(prefill_model, plugin, prefill_config));
     NPUW_ASSERT(m_prefill_compiled && "Can't create ov::npuw::CompiledModel for passed prefill "
                                       "model and its config, please check passed config.");
-
+    if (tail_mm_model) {
+        m_tail_mm_compiled_opt = std::dynamic_pointer_cast<ov::npuw::CompiledModel>(
+            ov::npuw::ICompiledModel::create(tail_mm_model, plugin,
+                ov::AnyMap{ov::intel_npu::npuw::partitioning::online::pipeline("NONE")}));
+        NPUW_ASSERT(m_tail_mm_compiled_opt.has_value() && m_tail_mm_compiled_opt.value() &&
+                    "Can't create ov::npuw::CompiledModel for found tail vocabulary matmul "
+                    "model.");
```

Comment:
This check doesn't seem to make sense. You just assigned to optional, it does have value.

Can `::create()` return a null ptr though? I though we're raising exceptions? Or the case is to have `dynamic_pointer_cast` be checked?

Reply:
Good point, thanks!

I doubt that these asserts are meaningful with written messages in them. Might be we can leave them here but fix messages to contain "cast failure"

Reply:
Hoewever, I guessed that they might be added to check that `dynamic_cast` returned `ov:npuw::CompiledModel` and not `ov::LLMCompiledModel` in case some one sent something like this `{{"++NPUW_LLM_GENERATE_CONFIG", ov::AnyMap {{"NPUW_LLM", "YES"}} }}`. But this is not the case for tail model, so removing assert here.

Reply:
However might be it is better to preserve assert in the case if `::create()` will be extended and may return not `ov::npuw::CompiledModel` as sanity check. But I'll remove confusing message here.